Add wildcard ignore for .configure-files folder

mokagio opened the pull request:
Makes sure that all the non-encrypted files in the `.configure-files` folder are ignored, for future proofing.

To test: Add a file the folder and verify it doesn't appear in the `git status` output.


## Regression Notes
N.A.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mokagio committed:
Add wildcard ignore for .configure-files folder

mokagio commented on .gitignore:
```diff
@@ -91,3 +91,8 @@ WordPress/debug.keystore
 
 local-builds.gradle
+
+# All secrets should be stored under .configure-files
```
After opening this PR, I realized that the wording here may be inappropriate. As a matter of fact, none of the secrets are currently stored in `.configure-files`.

🤔 

I still think there's value in encouraging developers to store all decrypted secrets in a single, _ignored_ location. Along the lines of https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1964.

oguzkocer replied:
Although I agree with the idea of putting properties under one folder, I am not sure if `.configure-files` folder is the right place for them. I still like the change as a future proof, but I'd prefer to merge it without the `All secrets should be stored under .configure-files` comment until we decide how exactly we want to go about this.

mokagio replied:
Thanks @oguzkocer. I've been thinking more about this since opening and I agree with you. 

Decoupling the folder in which we store the encrypted files from the one where we store the decrypted ones would make it more straightforward to ignore one but not the other. It would also make what's going on clearer for users with less context than infrastructure engineers, I think.

Here's an idea at the top of my head:

```
.configure/
  config.json # eventually, I'd like a YML because of support for comments
  encrypted/
    abc.kt.enc
  decrypted/ # this folder is ignored
    abc.kt
```

We could even have a `.gitignore` relative to the `.configure` folder which ignores the `decrypted` subfolder. But I may be getting ahead of myself...

mokagio replied:
As you mentioned, changing the folder structure is out of scope for this PR.

I'll update the wording in the `.gitignore` and request a new review from you. Thanks!

AliSoftware replied:
I love the suggestion of this new folder structure (even if indeed that's prob outside the scope of this PR, but I like it for our future plans)

mokagio committed:
Update wording of .configure-file ignored comment

https://github.com/wordpress-mobile/WordPress-Android/pull/14526/files#r619933550

mokagio commented:
@oguzkocer in `670442e`, I updated the wording of the comment as discussed [here](https://github.com/wordpress-mobile/WordPress-Android/pull/14526/files#r619933550). What do you think?

jkmassel reviewed (approved):
Let's land this for now, given the fact that _some_ files may live in this directory, and at the moment we're taking a better-safe-than-sorry approach.

We can address the adjusted directory structure down the road :)
